numpy/_core/src/umath/string_ufuncs.cpp

```diff
@@ -342,18 +342,24 @@ string_replace(Buffer<enc> buf1, Buffer<enc> buf2, Buffer<enc> buf3, npy_int64 c
     npy_int64 len1 = buf1.num_codepoints();
     npy_int64 len2 = buf2.num_codepoints();
     npy_int64 len3 = buf3.num_codepoints();
-
     Buffer<enc> end1 = buf1 + len1;
-    // Only try to replace if useful.
-    if (len1 >= len2  // Input is big enough to make a match possible.
-        && len2 > 0  // Match string is not empty (so output will be finite).
-        && !(len2 == len3 && buf2.strcmp(buf3) == 0)  // Match and replacement differ.
-        ) {
+
+    // Only try to replace if useful, otherwise just copy over data.
+    if (count <= 0                      // There's something to replace
```

Comment:
Unrelated to this PR but it's something I noticed elsewhere - can we make `num_codepoints` return a `size_t` or `npy_uint64`? That way we don't ever have to worry about somehow generating a negative number or UB from overflow.

Reply:
That's a good point. I'll open a separate PR for this as well.